cloud/azure/azure_rm_networkinterface.py

```diff
@@ -0,0 +1,619 @@
+        self.state = None
+        self.subnet_name = None
+        self.tags = None
+        self.virtual_network_name = None
+        self.security_group_name = None
+        self.os_type = None
+        self.open_ports = None
+        self.public_ip_allocation_method = None
+        self.public_ip = None
+
+        self.results = dict(
+            changed=False,
+            results=dict(),
+        )
+
+        super(AzureRMNetworkInterface, self).__init__(derived_arg_spec=self.module_arg_spec,
+                                                      supports_check_mode=True)
+
+    def exec_module(self, **kwargs):
+
+        for key in self.module_arg_spec.keys() + ['tags']:
+            setattr(self, key, kwargs[key])
+
+        self.results['check_mode'] = self.check_mode
+
+        results = dict()
+        changed = False
+        nic = None
+        subnet = None
+        nsg = None
+        pip = None
+
+        resource_group = self.get_resource_group(self.resource_group)
+        if not self.location:
+            # Set default location
+            self.location = resource_group.location
+
+        if not NAME_PATTERN.match(self.name):
+            self.fail("Parameter error: name must begin with a letter or number, end with a letter or number "
+                      "and contain at least one number.")
+
+        if self.state == 'present':
+            if self.virtual_network_name and not self.subnet_name:
+                self.fail("Parameter error: a subnet is required when passing a virtual_network_name.")
+
+            if self.subnet_name and not self.virtual_network_name:
+                self.fail("Parameter error: virtual_network_name is required when passing a subnet value.")
+
+            if self.virtual_network_name and self.subnet_name:
+                subnet = self.get_subnet(self.virtual_network_name, self.subnet_name)
+
+            if self.public_ip_address_name:
+                pip = self.get_public_ip_address(self.public_ip_address_name)
+
+            if self.security_group_name:
+                nsg = self.get_security_group(self.security_group_name)
+
+        try:
+            self.log('Fetching network interface {0}'.format(self.name))
+            nic = self.network_client.network_interfaces.get(self.resource_group, self.name)
+
+            self.log('Network interface {0} exists'.format(self.name))
+            self.check_provisioning_state(nic, self.state)
+            results = nic_to_dict(nic)
+            self.log(results, pretty_print=True)
+
+            if self.state == 'present':
+
+                update_tags, results['tags'] = self.update_tags(results['tags'])
+                if update_tags:
+                    changed = True
+
+                if self.private_ip_address:
+                    if results['ip_configuration']['private_ip_address'] != self.private_ip_address:
+                        self.log("CHANGED: network interface {0} private ip".format(self.name))
```

Comment:
Not sure what that means or what you're referring to. As far as I know, diff is not in scope for these modules.